python/cuspatial/cuspatial/geometry/geoarrowbuffers.py

```diff
@@ -0,0 +1,558 @@
+    * polygons_xy
+    * polygons_z
+    * polygons_polygons
+    * polygons_rings
+    * mpolygons
+
+    There are no correlations in length between any of the above columns.
+    Accepted host buffer object types include python list and any type that
+    implements numpy's `__array__` protocol.
+
+    Notes
+    -----
+    Legacy cuspatial algorithms depend on separated x and y columns. Access
+    them with the `.x` and `.y` properties.
+
+    Examples
+    --------
+    GeoArrowBuffers accept a dict as argument:
+
+    >>> buffers = GeoArrowBuffers({
+
+    >>> buffers = GeoArrowBuffers({
+            "points_xy": [0, 1, 2, 3],
+            "multipoints_xy": [0, 1, 2, 3],
+            "multipoints_offsets": [0, 1, 2],
+            "lines_xy": [0, 1, 2, 3],
+            "lines_offsets": [0, 1, 2],
+            "mlines": [1, 3],
+            "polygons_xy": [0, 1, 2, 3],
+            "polygons_polygonsets": [0, 1, 2],
+            "polygons_rings": [0, 1, 2],
+            "mpolygons": [1, 3],
+        })
+    """
+
+    def __init__(self, data: Union[dict, T], data_locale: object = cudf):
+        """
+        Parameters
+        ----------
+        data : A dict or a GeoArrowBuffers object.
+        """
+        if isinstance(data, dict):
+            if data.get("points_xy") is not None:
+                self._points = CoordinateArray(
+                    data["points_xy"], data_locale=data_locale
+                )
+            if data.get("mpoints_xy") is not None:
+                if data.get("mpoints_offsets") is None:
+                    raise ValueError("mpoints must include offsets buffer")
+                self._multipoints = MultiPointArray(
+                    data["mpoints_xy"],
+                    data["mpoints_offsets"],
+                    data_locale=data_locale,
+                )
+            if data.get("lines_xy") is not None:
+                if data.get("lines_offsets") is None:
+                    raise ValueError("lines_xy must include offsets buffer")
+                self._lines = LineArray(
+                    data["lines_xy"],
+                    data["lines_offsets"],
+                    data.get("mlines"),
+                    data_locale=data_locale,
+                )
+            if data.get("polygons_xy") is not None:
+                if (
+                    data.get("polygons_polygons") is None
+                    or data.get("polygons_rings") is None
+                ):
+                    raise ValueError(
+                        "polygons_xy must include polygons and" "rings buffers"
+                    )
+                self._polygons = PolygonArray(
+                    data["polygons_xy"],
+                    data["polygons_polygons"],
+                    data["polygons_rings"],
+                    data.get("mpolygons"),
+                    data_locale=data_locale,
+                )
+        elif isinstance(data, GeoArrowBuffers):
+            if hasattr(data, "_points"):
```

Comment:
As discussed, we might either want to use `data._points is not None` here; or:

```python
try:
    self._points = CoordinateArray(
                    data.points.xy, data.points.z, data_locale=data_locale
                )
except AttributeError:
    pass
```

Personally, I prefer the first here

Reply:
I'm leery about using `try:except:` in the way you've specified here. It is supposed to catch `AttributeError` from the ctor call alone, but if there are any nested `AttributeErrors` within it they'll be obscured and result in really wacky bugs that will be hard to debug.

I've changed it to use `if data.points is not None`, eliminating `hasattr`. If there's a problem with `data.points.xy` or data.points.z` they'll throw an exception, or if there's a problem in the constructor, instead of assuming that `data.points` is the only place I could see an `AttributeError`. Sound good?